### PR TITLE
Add Debian 11 support

### PR DESCRIPTION
**Pull request by kenyon**
Also includes #102.

### CODE REVIEW

**Comment by bastelfreak** [comment]
can you remove the RedHat mocking in https://github.com/puppetlabs/influxdb/blob/main/spec/classes/init_spec.rb#L4 ?

**Comment by kenyon** [comment]
I saw that and thought it looked wrong too. Removed.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 
 describe 'influxdb' do
-  let(:facts) { { os: { family: 'RedHat' }, identity: { user: 'root' } } }
+  let(:facts) { { identity: { user: 'root' } } }
```
Comment (bastelfreak) [inline spec/classes/init_spec.rb:4]: ah the following context blocks need to go in `on_supported_os.each do |os, os_facts|`

Comment (kenyon) [reply spec/classes/init_spec.rb:4]: Yep I realized that and am working on fixing it. The tests then fail due to some hardcoded values.

**Comment by kenyon** [comment]
@bastelfreak maybe there is a better way to handle the different operating systems in the spec tests, but this at least runs tests with the facts for each OS now!

**Comment by kenyon** [comment]
That AlmaLinux failure looks unrelated, let's try running the tests again...

**Comment by MartyEwings** [comment]
@m0dular we good to merge this?

**Review by m0dular — APPROVED** [review]
Works great, thanks :+1:

**Comment by m0dular** [comment]
I meant to only rerun the Alma test, but accidentally kicked them all off again.  I think we should be good to merge this regardless of that test, though.

**Pull request merged**